core/src/main/scala/org/broadinstitute/dsde/rawls/billing/BpmBillingProjectLifecycle.scala

```diff
@@ -134,11 +134,23 @@ class BpmBillingProjectLifecycle(
             )
             _ <- billingRepository.setBillingProfileId(createProjectRequest.projectName, profileModel.getId)
           } yield CreationStatuses.CreatingLandingZone).recoverWith { case t: Throwable =>
-            cleanupLandingZone(landingZone.getLandingZoneId, projectName, ctx) >> Future.failed(t)
+            logger.error("Billing project creation failed, cleaning up landing zone")
+            Option(landingZone.getLandingZoneId) match {
+              case Some(landingZoneId) => cleanupLandingZone(landingZoneId, projectName, ctx) >> Future.failed(t)
+              case _                   => Future.failed(t)
```

Comment:
Because we throw an Exception if the job report returned when initiating a landing zone has an error report in it, it is possible we can get in here and yet landingZone does not have a landingZoneId. In that case, we don't want to try to clean it up.